gift_compassion/models/contracts.py

```diff
@@ -47,10 +47,22 @@ def invoice_paid(self, invoice):
             if invl.product_id.categ_name == GIFT_CATEGORY and \
                     invl.contract_id.child_id:
                 # Create the Sponsorship Gift
-                self.env['sponsorship.gift'].create_from_invoice_line(invl)
+                gift = self.env['sponsorship.gift']\
+                    .create_from_invoice_line(invl)
+
+                if not invl.invoice_id.is_active:
+                    gift.state = 'verify'
 
         super(SponsorshipContract, self).invoice_paid(invoice)
 
+    @api.onchange('is_active')
```

Comment:
You cannot use api.onchange method to listen to a change in another object. Onchange decorator is meant only to listen to changes in a form view ([see documentation here](https://www.odoo.com/documentation/10.0/reference/orm.html#module-odoo.api))

The correct way is to go in the contract object and in the method `contract_active`, searching for any pending gifts related to the contract and write them in state draft.